engine/access/state_stream/account_status_filter.go

```diff
@@ -0,0 +1,212 @@
+		// Therefore, event types are validated with `validateCoreEventTypes` to fail at the beginning of filter creation.
+		err := validateCoreEventTypes(eventTypes)
+		if err != nil {
+			return AccountStatusFilter{}, err
+		}
+	} else if len(accountAddresses) > DefaultMaxAccountAddresses {
+		// If `accountAddresses` exceeds the `DefaultAccountAddressesLimit`, it returns an error.
+		return AccountStatusFilter{}, fmt.Errorf("account limit exceeds, the limit is %d", DefaultMaxAccountAddresses)
+	}
+
+	// If `eventTypes` is empty, the filter returns all core events for any accounts.
+	if len(eventTypes) == 0 {
+		eventTypes = DefaultCoreEvents
+	}
+
+	//  It's important to only set eventTypes if there are no addresses passed.
+	var filterEventTypes []string
+	if len(accountAddresses) == 0 {
+		filterEventTypes = eventTypes
+	}
+
+	// Creates an `EventFilter` with the provided `eventTypes`.
+	filter, err := NewEventFilter(config, chain, filterEventTypes, []string{}, []string{})
+	if err != nil {
+		return AccountStatusFilter{}, err
+	}
+
+	accountStatusFilter := AccountStatusFilter{
+		EventFilter: &filter,
+	}
+
+	for _, address := range accountAddresses {
+		// Validate account address
+		addr := flow.HexToAddress(address)
+		if err := validateAddress(addr, chain); err != nil {
+			return AccountStatusFilter{}, err
+		}
+
+		// If there are non-core event types at this stage, it returns an error from `addCoreEventFieldFilter`.
+		for _, eventType := range eventTypes {
+			// use the hex with prefix address to make sure it will match the cadence address
+			err = accountStatusFilter.addCoreEventFieldFilter(flow.EventType(eventType), addr.HexWithPrefix(), chain)
+			if err != nil {
+				return AccountStatusFilter{}, err
+			}
+		}
+	}
+
+	// We need to set hasFilters here if filterEventTypes was empty
+	accountStatusFilter.hasFilters = len(accountStatusFilter.EventFieldFilters) > 0 || len(eventTypes) > 0
+
+	return accountStatusFilter, nil
+}
+
+// GroupCoreEventsByAccountAddress extracts account-related core events from the provided list of events.
+// It filters events based on the account field specified by the event type and organizes them by account address.
+// Parameters:
+// - events: The list of events to extract account-related core events from.
+// - log: The logger to log errors encountered during event decoding and processing.
+// Returns:
+//   - A map[string]flow.EventsList: A map where the key is the account address and the value is a list of
+//     account-related core events associated with that address.
+func (f *AccountStatusFilter) GroupCoreEventsByAccountAddress(events flow.EventsList, log zerolog.Logger) map[string]flow.EventsList {
+	allAccountProtocolEvents := make(map[string]flow.EventsList)
+
+	for _, event := range events {
+		fields, fieldValues, err := getEventFields(&event)
+		if err != nil {
+			log.Info().Err(err).Msg("could not get event fields")
+			continue
+		}
+
+		accountField := f.EventFieldFilters[event.Type]
```

Comment:
I did some testing on localnet, and when no addresses are passed, this returns no results. I think that's because in that situation, we're not using `EventFieldFilters` so this method returns an empty map.

Reply:
> I did some testing on localnet, and when no addresses are passed, this returns no results. I think that's because in that situation, we're not using `EventFieldFilters` so this method returns an empty map.

you are right, added a test case for that and it failed. Thinking about the best approach for solving this.